script.js

```diff
@@ -1,12 +1,547 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+//Setting global variables
+
+let allShows;
+let allEpisodes;
+let isShowsListing = true;
+let episodeSelectContainer;
+let searchContainer;
+let showSearchInput;
+let showSelect;
+let searchInput;
+let episodeSelect;
+let showSearchContainer;
+
+
+
+async function setup() {
+  allShows = await getAllShows();
+  const selectedShowId = allShows[0].id; // Selecting the first show by default
+  allEpisodes = await getEpisodes(selectedShowId);
+  createHeader();
+  createShowSearch();
+  showSelectField(allShows);
+  showListing();
+  createFooter();
+}
+
+// Fetching episodes for shows
+function getEpisodes(showId) {
+  return fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+    .then(response => {
+      if (!response.ok) {
+        throw new Error('Error fetching episodes');
+      }
+      return response.json();
+    })
+    .catch(error => {
+      console.error('Error fetching episodes:', error);
+    });
+}
+
+
+
+
+function searchShows(searchValue, showSelect) {
+  let searchResults;
+
+  if (searchValue !== "") {
+    searchResults = allShows.filter((show) => {
+      // Search through show names, genres, and summary texts
+      const { name, genres, summary } = show;
+      const lowerCaseName = name.toLowerCase();
+      const lowerCaseGenres = genres.join(", ").toLowerCase();
+      const lowerCaseSummary = summary.toLowerCase();
+
+      return (
+        lowerCaseName.includes(searchValue) ||
+        lowerCaseGenres.includes(searchValue) ||
+        lowerCaseSummary.includes(searchValue)
+      );
+    });
+
+    addSelectOption(showSelect, "Select Show");
+
+  } else {
+    searchResults = allShows;
+  }
+
+  while (showSelect.firstChild) {
+    showSelect.removeChild(showSelect.firstChild);
+  }
+
+  searchResults.forEach((show) => {
```

Comment:
So, you create options, cool. The question is, do we really need to display the founded show in the options?

Reply:
that's what i understood from the pic on level 500 but i'm not sure tbh :/